packages/git/src/browser/git-frontend-contribution.ts

```diff
@@ -4,19 +4,40 @@
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
-
 import { injectable, inject } from "inversify";
+import { GitRepositoryProvider } from './git-repository-provider';
 import { FrontendApplication, FrontendApplicationContribution } from '@theia/core/lib/browser';
 import { WidgetManager } from '@theia/core/lib/browser/widget-manager';
+import { StatusBar, StatusBarAlignment } from "@theia/core/lib/browser/statusbar/statusbar";
+import { Git } from '../common';
 
 export const GIT_WIDGET_FACTORY_ID = 'git';
 
 @injectable()
 export class GitFrontendContribution implements FrontendApplicationContribution {
     constructor(
-        @inject(WidgetManager) protected readonly widgetManager: WidgetManager
+        @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
+        @inject(GitRepositoryProvider) protected readonly repositoryProvider: GitRepositoryProvider,
+        @inject(Git) protected readonly git: Git,
+        @inject(StatusBar) protected readonly statusbar: StatusBar
     ) { }
 
+    onStart(app: FrontendApplication) {
+        this.repositoryProvider.onDidChangeRepository(async currentRepo => {
```

Comment:
Only updating when the repository is changed, is not enough. We want to update when the branch changes, too. For that it should use gitWatcher and react on `GitStatusChangeEvent`.